### PR TITLE
Added option to support custom shared libraries with <uses-library> tag

### PR DESCRIPTION
**Pull request by pax0r**
Support for: 
https://developer.android.com/guide/topics/manifest/uses-library-element

### CODE REVIEW

File: pythonforandroid/bootstraps/common/build/build.py [hunk 1]
```diff
@@ -749,6 +751,9 @@ def _read_configuration():
     if args.meta_data is None:
         args.meta_data = []
 
+    if args.android_used_libs is None:
+        args.android_used_libs = []
```
Comment (inclement) [inline pythonforandroid/bootstraps/common/build/build.py]: Very minor thing: is it enough to just write `default=[]` in the `ap.add_argument(...)` instead?

Comment (pax0r) [reply pythonforandroid/bootstraps/common/build/build.py]: I think it should be, I just wanted it to be consistent with existing code. If you like it I can refactor rest of it to use defaults.

Comment (pax0r) [reply pythonforandroid/bootstraps/common/build/build.py]: fixed

File: pythonforandroid/bootstraps/common/build/build.py [hunk 2]
```diff
@@ -740,18 +742,9 @@ def _read_configuration():
               'deprecated and does nothing.')
         args.sdk_version = -1  # ensure it is not used
 
-    if args.permissions is None:
-        args.permissions = []
-    elif args.permissions:
-        if isinstance(args.permissions[0], list):
+    if args.permissions and isinstance(args.permissions[0], list):
             args.permissions = [p for perm in args.permissions for p in perm]
```
Comment (AndreMiras) [inline pythonforandroid/bootstraps/common/build/build.py]: Thanks for updating that PR so many times and also thanks for squashing your commits!
We have one more here:
`pythonforandroid/bootstraps/common/build/build.py:746:13: E117 over-indented`
Don't kill us now :grimacing: 
You can always run `tox` locally on your dev env to verify test and linting is OK if you don't want to wait for Travis to complain

Comment (pax0r) [reply pythonforandroid/bootstraps/common/build/build.py]: Sorry, is obviously my fault for not running pep8 locally ;)  I mostly rely on pycharm for catching formatting errors but it failed me at this one ;)

**Comment by AndreMiras** [comment]
@inclement you have some ongoing "requested changes" that I think got addressed. Do you mind doing a final check and merging?

**Comment by AndreMiras** [comment]
@pax0r since @inclement seems to be afk I think we can merge soon. Have you been using this change for a while without issues? Do you think it's OK for a merge now?

**Comment by pax0r** [comment]
Yes, we build application using some 3rd party library and had no issues. After merge I'll also provide patch for buildozer with support for this option.

**Comment by AndreMiras** [comment]
Merged thanks!
@inclement comments were addressed, but the his review was still in "request changes" and that was blocking the merge, so I had to dismiss them

**Pull request merged**